npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -476,6 +499,71 @@ def self.dependencies_with_all_versions_metadata(dependency_set)
           dependency
         end
       end
+
+      # Fetch the registry URL for a given package manager
+      sig do
+        params(
+          name: String,
+          lockfiles: T::Hash[Symbol, T.nilable(Dependabot::DependencyFile)],
+          rc_files: T::Hash[Symbol, T.nilable(Dependabot::DependencyFile)]
+        ).returns(String)
+      end
+      def self.find_registry(name, lockfiles, rc_files)
+        find_registry_from_lockfile_or_rc(
+          lockfiles[name.to_sym],
+          rc_files[name.to_sym],
+          default: DEFAULT_REGISTRIES[name]
+        )
+      end
+
+      # Helper to determine registry from lockfile or RC file
+      sig do
+        params(
+          lockfile: T.nilable(Dependabot::DependencyFile),
+          rc_file: T.nilable(Dependabot::DependencyFile),
+          default: String
+        ).returns(String)
+      end
+      def self.find_registry_from_lockfile_or_rc(lockfile, rc_file, default:)
+        # Attempt to find registry in the lockfile
+        lockfile_content = lockfile&.content
+        if lockfile_content
+          lockfile_registry = parse_lockfile_for_registry(lockfile_content)
+          return lockfile_registry if lockfile_registry
+        end
+
+        # Attempt to find registry in the RC file
+        rc_file_content = rc_file&.content
+        if rc_file_content
+          rc_registry = parse_rc_file_for_registry(rc_file_content)
+          return rc_registry if rc_registry
+        end
+
+        # Fallback to the default registry
+        default
+      end
+
+      # Parse the lockfile to extract the registry URL
+      sig { params(content: String).returns(T.nilable(String)) }
+      def self.parse_lockfile_for_registry(content)
+        content.lines.map(&:strip).select { |line| line.start_with?("resolved ") }.filter_map do |line|
```

Comment:
why the empty space after resolved i.e. "resolved "

Reply:
I will check but all package manager keeps resolved like 


```
resolved https://registery-url